src/v/kafka/client/client.cc

```diff
@@ -117,27 +117,24 @@ ss::future<> client::stop() noexcept {
 ss::future<> client::update_metadata(wait_or_start::tag) {
     return ss::try_with_gate(_gate, [this]() {
         vlog(kclog.debug, "updating metadata");
-        return _brokers.any()
-          .then([this](shared_broker_t broker) {
-              return broker->dispatch(metadata_request{.list_all_topics = true})
-                .then([this](metadata_response res) {
-                    // Create new seeds from the returned set of brokers if
-                    // they're not empty
-                    if (!res.data.brokers.empty()) {
-                        std::vector<net::unresolved_address> seeds;
-                        seeds.reserve(res.data.brokers.size());
-                        for (const auto& b : res.data.brokers) {
-                            seeds.emplace_back(b.host, b.port);
-                        }
-                        std::swap(_seeds, seeds);
-                    }
-
-                    return apply(std::move(res));
-                })
-                .finally([]() { vlog(kclog.trace, "updated metadata"); });
-          })
-          .handle_exception_type(
```

Comment:
Nice. For this type of stuff we usually break it into two commits, one without formatting, then one with.

Reply:
Oh I run formatting by habit every time I change a file (I hate getting lint CI errors).